### PR TITLE
Move off of project.json onto .csproj

### PR DESCRIPTION
**Pull request by DustinCampbell**
This is significant change that moves omnisharp-roslyn from project.json to csproj. This means that, if anyone is using Visual Studio to develop OmniSharp must use VS 2017, which became generally available just yesterday.

There are a lot of changes in the build script related to building on macOS/Linux. With the latest .NET Core SDK, building a .NET Framework TFM (like net46) on macOS/Linux is no longer supported. So, we'll use the msbuild that ships as part of Mono 4.8. Note that anyone building on Linux must also install the msbuild package.

### CODE REVIEW

File: src/OmniSharp.Plugins/OmniSharp.Plugins.csproj [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+  <ItemGroup>
+    <ProjectReference Include="..\OmniSharp.Abstractions\OmniSharp.Abstractions.csproj" />
+  </ItemGroup>
+
+  <ItemGroup>
+    <PackageReference Include="Newtonsoft.Json" Version="9.0.1" />
+  </ItemGroup>
+
+  <ItemGroup Condition=" '$(TargetFramework)' == 'net46' ">
+    <Reference Include="System" />
```
Comment (david-driscoll) [inline src/OmniSharp.Plugins/OmniSharp.Plugins.csproj]: Are these references required?  I haven't needed them for any of my cross targeting stuff yet (but I'm not running any of my stuff with mono at the moment).

File: BUILD.md [hunk 2]
```diff
@@ -1,29 +1,54 @@
...
-  
-  `-install-path=Path`: Path used for the **Install** target.
+
+  `-install-path Path`: Path used for the **Install** target.
     Defaults to `(%USERPROFILE%|$HOME)/.omnisharp/local`
-  
+
   `-archive`: Enable the generation of publishable archives after a build.
 
+Note: On macOS/Linux, be sure to pass the arguments above with double slashes! (e.g. `--target TargetName`).
```
Comment (filipw) [inline BUILD.md]: is this how cake behaves? very non-intuitive

Comment (DustinCampbell) [inline BUILD.md]: I sort of agree with you, but I get why it's done that way. Single dashes are the general behavior of PowerShell, while double dashes are more appropriate on Unix. This is how the cake scripts work for [PowerShell](https://github.com/cake-build/example/blob/master/build.ps1) and [Bash](https://github.com/cake-build/example/blob/master/build.sh).

**Comment by filipw** [comment]
do we still need https://github.com/OmniSharp/omnisharp-roslyn/blob/dev/NuGet.Config#L7?  Roslyn is pulled from the public nuget feed now

File: src/OmniSharp.Plugins/OmniSharp.Plugins.csproj (hunk 1 above)
Comment (DustinCampbell) [inline src/OmniSharp.Plugins/OmniSharp.Plugins.csproj]: You know, I'm not sure. These got added when running `dotnet migrate`. Let me ask around and I'll remove them if they aren't needed.

File: BUILD.md (hunk 2 above)
Comment (david-driscoll) [inline BUILD.md]: I've actually modified my cake scripts to have the first argument be the traget, much like grunt / gulp.

I don't have a shell version yet however.

File: src/OmniSharp.Plugins/OmniSharp.Plugins.csproj (hunk 1 above)
Comment (david-driscoll) [inline src/OmniSharp.Plugins/OmniSharp.Plugins.csproj]: Yeah, they were added for me, but I was able to safely remove them and have not had any problems.  Keep in mind I'm not running on mac at this point however.

File: BUILD.md (hunk 2 above)
Comment (DustinCampbell) [inline BUILD.md]: That's a very good idea! Although, if we ever update the Cake scripts, we'll need to remember to keep any modifications.

**Comment by DustinCampbell** [comment]
@filipw:

> do we still need https://github.com/OmniSharp/omnisharp-roslyn/blob/dev/NuGet.Config#L7? Roslyn is pulled from the public nuget feed now

Good point. I'll pull that out.

File: src/OmniSharp.Plugins/OmniSharp.Plugins.csproj (hunk 1 above)
Comment (DustinCampbell) [inline src/OmniSharp.Plugins/OmniSharp.Plugins.csproj]: OK. I got the answer I was looking for. Awhile ago, these references were necessary but they are now added implicitly in the SDK. I'll go ahead and remove them.

Comment (david-driscoll) [inline src/OmniSharp.Plugins/OmniSharp.Plugins.csproj]: 👍

**Comment by DustinCampbell** [comment]
OK. Feedback should be addressed. Thanks for taking a look!

**Comment by david-driscoll** [comment]
🎉🎈🎉🎈🎉🎈🎉🎈🎉🎈🎉

**Comment by david-driscoll** [comment]
🍻 Go home travis you're drunk 🍻

**Comment by filipw** [comment]
:shipit:

**Comment by DustinCampbell** [comment]
Boom!

**Comment by davidfowl** [comment]
👏

**Pull request merged**